gocourse04/smart_zoo_test.go

```diff
@@ -0,0 +1,218 @@
+}
+
+func TestDeleteArea(t *testing.T) {
+	zoo := NewZoo("Happy zoo")
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	zoo.AddArea(area)
+
+	_, exists := zoo.GetArea("area_1")
+	assert.True(t, exists, "Area should exist")
+
+	zoo.DeleteArea("area_1")
+	_, exists = zoo.GetArea("area_1")
+	assert.False(t, exists, "Area shouldn't exist")
+}
+
+func TestLookupAnimalByID(t *testing.T) {
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	sector := NewSector("sector_1", "Parrots", nil)
+	sector2 := NewSector("sector_2", "Ducks", nil)
+	animal := NewAnimal("animal_1", "Frodo")
+	animal2 := NewAnimal("animal_2", "Bilbo")
+	sector.AddAnimal(animal)
+	sector.AddAnimal(animal2)
+	area.AddSector(sector)
+	area.AddSector(sector2)
+	zoo := NewZoo("Happy zoo")
+	zoo.AddArea(area)
+
+	sectorFound, ok := zoo.LookupAnimalByID("animal_2")
+
+	assert.True(t, ok, "Animal should be found")
+	assert.Equal(t, sectorFound, sector, "Sectors should be equal")
+}
+
+func TestFailLookupAnimalByID(t *testing.T) {
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	sector := NewSector("sector_1", "Parrots", nil)
+	sector2 := NewSector("sector_2", "Ducks", nil)
+	animal := NewAnimal("animal_1", "Frodo")
+	animal2 := NewAnimal("animal_2", "Bilbo")
+	sector.AddAnimal(animal)
+	sector.AddAnimal(animal2)
+	area.AddSector(sector)
+	area.AddSector(sector2)
+	zoo := NewZoo("Happy zoo")
+	zoo.AddArea(area)
+
+	sectorFound, ok := zoo.LookupAnimalByID("animal_3")
+
+	assert.False(t, ok, "Animal should be found")
+	assert.Equal(t, sectorFound, Sector{}, "Sectors should be equal")
+}
+
+func TestLookupAnimalByName(t *testing.T) {
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	sector := NewSector("sector_1", "Parrots", nil)
+	sector2 := NewSector("sector_2", "Ducks", nil)
+	animal := NewAnimal("animal_1", "Frodo")
+	animal2 := NewAnimal("animal_2", "Bilbo")
+	animal3 := NewAnimal("animal_3", "Frodo")
+	sector.AddAnimal(animal)
+	sector.AddAnimal(animal2)
+	sector2.AddAnimal(animal3)
+	area.AddSector(sector)
+	area.AddSector(sector2)
+	zoo := NewZoo("Happy zoo")
+	zoo.AddArea(area)
+
+	sectorsFound, ok := zoo.LookupAnimalByName("Frodo")
+
+	assert.True(t, ok, "Animals should be found")
+	assert.Len(t, sectorsFound, 2, "Should be 2 sectors")
+}
+
+func TestFailLookupAnimalByName(t *testing.T) {
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	sector := NewSector("sector_1", "Parrots", nil)
+	sector2 := NewSector("sector_2", "Ducks", nil)
+	animal := NewAnimal("animal_1", "Frodo")
+	animal2 := NewAnimal("animal_2", "Bilbo")
+	animal3 := NewAnimal("animal_3", "Frodo")
+	sector.AddAnimal(animal)
+	sector.AddAnimal(animal2)
+	sector2.AddAnimal(animal3)
+	area.AddSector(sector)
+	area.AddSector(sector2)
+	zoo := NewZoo("Happy zoo")
+	zoo.AddArea(area)
+
+	sectorsFound, ok := zoo.LookupAnimalByName("Gaga")
+
+	assert.False(t, ok, "Animals shouldn't be found")
+	assert.Len(t, sectorsFound, 0, "Should be 0 sectors")
```

Comment:
Можна ось так спростити:
```suggestion
	assert.Empty(t, sectorsFound, "Sectors should be empty")
```